tests/parser_test.py

```diff
@@ -427,3 +428,98 @@ def test_keyword_identifiers(text: str, rule: str) -> None:
     """
     unit = parse_buffer(text, rule=rule)
     assert len(unit.diagnostics) == 0, text + "\n".join(str(d) for d in unit.diagnostics)
+
+
+@pytest.mark.parametrize(
+    "string,expected",
+    [
+        (
+            "(case A is when V1 => 1, when V2 => 2)",
```

Comment:
How about also testing a qualified literal as a selecting expression and a boolean/mathematical expression as dependent expression?

Reply:
There are still just test cases with numeric literals as dependent/choice expression. Shouldn't we also add a test with a more complex dependent/choice expression?

Reply:
Added test cases with numeric/boolean expression.